server/src/main/kotlin/org/javacs/kt/codeaction/quickfix/QuickFix.kt

```diff
@@ -16,10 +16,18 @@ interface QuickFix {
 }
 
 fun diagnosticMatch(diagnostic: Diagnostic, range: Range, diagnosticTypes: Set<String>): Boolean =
-    diagnostic.range.equals(range) && diagnosticTypes.contains(diagnostic.code.left)
+    isDiagnosticInRange(diagnostic, range) && diagnosticTypes.contains(diagnostic.code.left)
+
+// for a diagnostic to be in range the lines should be the same, and
+//  the input character range should be within the bounds of the diagnostics range.
+private fun isDiagnosticInRange(diagnostic: Diagnostic, range: Range): Boolean {
+    val diagnosticRange = diagnostic.range
+    return diagnosticRange.start.line == range.start.line && diagnosticRange.end.line == range.end.line &&
+        diagnosticRange.start.character <= range.start.character && diagnosticRange.end.character >= range.end.character
+}
 
 fun diagnosticMatch(diagnostic: KotlinDiagnostic, startCursor: Int, endCursor: Int, diagnosticTypes: Set<String>): Boolean =
-    diagnostic.textRanges.any { it.startOffset == startCursor && it.endOffset == endCursor } && diagnosticTypes.contains(diagnostic.factory.name)
+    diagnostic.textRanges.any { it.startOffset <= startCursor && it.endOffset >= endCursor } && diagnosticTypes.contains(diagnostic.factory.name)
```

Comment:
Similarly here, it might be worth checking if there is a similar function for Kotlin's `TextRange`, this might be useful in other places too.

Reply:
I think I will look further into that in a future PR. Right now I don't see how it will improve the current code without some rewrites. I might be wrong though 🙂